File: Models/Quote.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace web_agent_pro.Models
{
    public class Quote 
    {
      public Quote()
        {
            Drivers = new HashSet<Driver>();
            Vehicles = new HashSet<Vehicle>();
        }

        [DisplayName("Quote ID")]
        public long Id { get; set; }

        [Required]
        [DisplayName("User Id")]
        [MaxLength(450)]
        public string UserId { get; set; }
        
        [Required]
        [DisplayName("First Name")]
        [MaxLength(50)]
        public string FirstName { get; set; }

        [Required]
        [DisplayName("Last Name")]
        [MaxLength(50)]
        public string LastName { get; set; }

        [Required]
        [DisplayName("Date of Birth")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime DateOfBirth { get; set; }

        [Required]
        public string Email { get; set; }

        public string Phone { get; set; }

        [Required]
        [MaxLength(40)]
        public string Address { get; set; }

        [Required]
        [MaxLength(50)]
        public string City { get; set; }

        [Required]
        public string State { get; set; }

        [Required]
        [StringLength(5, MinimumLength = 5)]
        public string Zip { get; set; }

        [Required]
        [DisplayName("Social Security Number")]
        [StringLength(9, MinimumLength = 9)]
        public string SSN { get; set; }

        [Required]
        [DisplayName("Previous Carrier")]
        public string PreviousCarrier { get; set; }

        [DisplayName("Claims in last 5 years")]
        public bool PastClaims { get; set; }

        public double? PastClaimsDiscount { get; set; }

        [DisplayName("Any moving violation in last 5 years")]
        public bool MovingViolations { get; set; }

        public double? MovingViolationsDiscount { get; set; }

        [DisplayName("Less than 3 years driving")]
        public bool NewDriver { get; set; }

        [DisplayName("New Driver Discount")]
        public double? NewDriverDiscount { get; set; }

        [DisplayName("Force multi-car discount")]
        public bool MultiCar { get; set; }

        [DisplayName("Multi-car Discount")]
        public double? MultiCarDiscount { get; set; }

        [DisplayName("Previous Carrier is Lizard")]
        public bool PreviousCarrierLizard { get; set; }

        public double? PreviousCarrierLizardDiscount { get; set; }

        [DisplayName("Previous Carrier is Pervasive")]
        public bool PreviousCarrierPervasive { get; set; }

        public double? PreviousCarrierPervasiveDiscount { get; set; }

        [DisplayName("Price")]
        public decimal Price { get; set; }

        [DisplayName("Date Quoted")]
        public DateTime DateQuoted { get; set; }

        [DisplayName("Submitted?")]
        public bool Submitted { get; set; }

        [DisplayName("Date Sold")]
        public DateTime? DateSold { get; set; }

        [DisplayName("Discarded")]
        public bool Discarded { get; set; }

        [DisplayName("Date Discarded")]
        public DateTime? DateDiscarded { get; set; }
        public decimal QuoteMultiplier { get; set; }

        public virtual ApplicationUser User { get; set; }
        
        public ICollection<Driver> Drivers { get; set; }

        public ICollection<Vehicle> Vehicles { get; set; }
    }
}

File: ClientApp/src/data/models/domain/quote.ts
import { Driver } from "./driver";
import { Vehicle } from "./vehicle";

export class Quote {
  constructor() {}

  id: number;
  userId: string;
  firstName: string;
  lastName: string;
  dateOfBirth: Date;
  email: string;
  phone: string;
  address: string;
  city: string;
  state: string;
  zip: string;
  ssn: string;
  price: number;
  dateQuoted: Date;
  submitted: boolean;
  dateSold: Date;
  discarded: boolean;
  dateDiscarded: Date;
  previousCarrier: string;
  pastClaims: boolean;
  pastClaimsDiscount: number;
  movingViolations: boolean;
  movingViolationsDiscount: number;
  newDriver: boolean;
  newDriverDiscount: number;
  multiCar: boolean;
  multiCarDiscount: number;
  previousCarrierLizard: boolean;
  previousCarrierLizardDiscount: number;
  previousCarrierPervasive: boolean;
  previousCarrierPervasiveDiscount: number;
  drivers: Driver[];
  vehicles: Vehicle[];
  quoteMultiplier: number;

  addDriver(driver: Driver): void {
    this.drivers.push(driver);
  }
  
  updateDriver(driver: Driver): void {
    let driverToUpdate = this.drivers.find(d => d.id == driver.id);
    if(!driverToUpdate) return;

    let updated = Object.assign(driverToUpdate, driver);
  }

  deleteDriver(driver: Driver | number): void {
    const id = typeof driver === 'number' ? driver : driver.id;
    this.drivers = this.drivers.filter(d => d.id != id);
  }

  addVehicle(vehicle: Vehicle): void {
    this.vehicles.push(vehicle);
  }

  updateVehicle(vehicle: Vehicle): void {
    let vehicleToUpdate = this.vehicles.find(v => v.id == vehicle.id);
    if(!vehicleToUpdate) return;

    let updated = Object.assign(vehicleToUpdate, vehicle);
  }

  deleteVehicle(vehicle: Vehicle | number): void {
    const id = typeof vehicle === 'number' ? vehicle : vehicle.id;
    this.vehicles = this.vehicles.filter(v => v.id != id);
  }

}
